source/pakker.rst

```diff
@@ -3,3 +3,214 @@
 Gode Pythonpakker
 =================
 
+I dette kapitel beskrive en række gode og anbefalelsesværdige Pythonpakker, der kan hjælpe med at
+løse gængse problemstillinger en udvikler i SDFE ofte vil støde på. I udvælgelsen af pakkerne
+herunder er der lagt vægt på særligt pythonske implementeringer, der benytter sig af koncepter
+introduceret i nyere udgaver af Python.
+
+Håndtering af geodata
+----------------------------
+
+`GDAL`_ er det mest udbredte programmel til læsning, skrivning og manipulation
+af geodata. GDAL kan bearbejde både raster- og vektordata. Vektordata manipuleres med
+underbiblioteket OGR. GDAL-projektet tilbyder pythonpakke men den er ikke særligt intuitiv at
+bruge for pythonudviklere, da der er tale om en en-til-en mapning af det underliggende C API.
+Bedre alternativer er pakkener Rasterio (rasterdata) og Fiona (vektordata), der begge tilbyder
+et bedre og mere pythonsk API til GDAL og OGR.
+
+
+Rasterio
++++++++++
+
+`Rasterio dokumentation`_.
+
+Rasterio bruges til at læse og skrive rasterdata. Herunder eksempler på begge dele.
+
+Læsning af rasterdata er utroligt simpelt og minder meget om måden almindelige tekstfiler
+læses med Python:
+
+.. code-block:: python
+
+   import rasterio
+
+    with rasterio.open("eksempel.tif", "r") as grid:
+        # Læs information om hele filen
+        epsg_kode = grid.crs
+        gridstørrelse = grid.shape
+
+        # Behandl data for hvert enkelt bånd i rasterfilen.
+        # Bemærk at data returnes i form af numpy arrays.
+        for båndnr in grid.indexes:
+            data = grid.read(båndnr)
+            gennemsnit = data.mean()
+
+Oprettelse af en ny fil og eftefølgende skrivning af nyt data er mere omfattende. Det forudsættes
+at der er taget stilling til filens dækningsområde, koordinatsystem, antal bånd, gridstørrelse,
+filformat og datatype. Alt dette kan dog klares relativt simpelt, som demonstreret i eksemplet
+herunder, hvor vi laver en tiff-fil med tre bånd, der omtrentligt dækker det danske område og
+registrerer data i ETRS89-koordinater.
+
+.. code-block:: python
+
+    import rasterio
+    import numpy as np
+
+    # Grid dimensioner
+    antal_bånd = 3
+    højde = 256
+    bredde = 256
+
+    # Dækningsområde
+    vest = 8.0
+    øst = 13.0
+    syd = 54.0
+    nord = 58.0
+
+    # Opsætning af rasterfil
+    kwargs = dict(
+        driver="GTiff",
+        height=højde,
+        width=bredde,
+        count=antal_bånd,
+        crs="EPSG:4258", # ETRS89, geografiske koordinater
+        dtype=data.dtype,
```

Comment:
Det var ikke klart for mig, at eksemplet fortsatte med `data`, som bliver erklæret ovenfor. Det fremstod som et separat eksempel, indtil jeg så, at der stod `data.dtype` her.

Jeg tror, at min forvirring kom af måden teksten læger op til eksemplet, og at der bliver importeret påny i toppen af dette eksempel. I teksten kunne der måske stå, at vi bygger videre på det forrige.

Reply:
Ja, `data` fra linje 82 skal bare flyttes op over definition af `kwargs`, så virker det som et selvstående eksempel. Hvilket var hensigten.

Reply:
Jeg endte med at eksplicit angive datatype i `kwargs`. Så kører eksemplet fint.